### ISSUE TITLE
Binary performance optimizations

### ISSUE THREAD

**Issue by sean-mcmanus** [microsoft/vscode-cpptools#9667]
UPDATE: We're looking into getting better performance tests first, which will be tracked in some other issue.
- [ ] Linux
  - done, but % improvement unknown yet -- need to add a performance test
- [ ] Mac
  - need to investigate a build failure
- [x] Windows
  - Minor optimizations done.
  - Further optimizations postponed.

**Comment by jwsv61099** [comment]
Been following the CPP Tools releases closely and have seen this issue moved around. Was curious if we could get some more information about this issue?

**Comment by sean-mcmanus** [comment]
Yeah, this task keeps getting pushed back in the schedule due to higher priority work (feature work, bug fixes, etc.).

We've implemented binary performance optimizations for Linux (Mac was started but hit an issue that needs to be resolved), but we're not sure yet if it makes the performance better or worse (we've seen both, depending on certain variations in the build), so we're planning to add better performance testing first.

We already shipped some minor Windows optimization (like 5%), but the additional binary performance optimization didn't yield a detectable benefit (possibly due to some misconfiguration in the optimization process), so additional work on that has been postponed and I'm not sure if/when it'll be revisited.
